### ISSUE TITLE
Issues resolving absolute paths in <a href>

### ISSUE THREAD

**Issue by samholmes** [multiformats/multiformats#40]
As it stands, the multiaddr protocol does not specify whether an address is domain specific. There is no way to tell whether an anchor element's (`<a>`) href attribute is an absolute reference to a resource or an relative path to the exiting domain. For example, given a webpage with the hash <page-hash> and a link with a hash <link-hash>: 

```
<a href='/ipfs/<link-hash>/path'>...</a>
```

How is a web browser (client) going to differentiate the resolution of this href from meaning either A or B:

A
```
/ipfs/<link-hash>/path
```

or B
```
/ipfs/<page-hash>/ipfs/<hash>/path
```

In conclusion, the multiaddr format cannot support paths relative to a webpage's domain. There simply is not concept of a webpage domain.

Please provide further commentary on the subject matter explaining the intentions to solve such an issue with the multiaddr proposal.

Thanks! :)

**Comment by samholmes** [comment]
What is the vision for multiaddr? Is it meant to work within the a URI, specifically the path of a URI? Or is it meant to be a better replacement, or an alternative to URI? Is it meant to work along side URI?

The reason why I want to discuss this is to understand the correct way to implement multiaddr into my projects that is aligned with the overall vision of multiaddr's purpose.

I'm currently experimenting with a web browser project that will somehow understand multiaddr addresses natively. 

My initial idea was to implement multiaddr into a URI scheme like `i://ipfs/<hash>`, but the challenge with this is that the domain for the web application would incorrectly become "ipfs" instead of the "<hash>". As fair as I know, the current implementation of the URI protocol provides no way for me to specify the domain segment of my URI scheme (Election's protocols API). This leaves me with two options:

1. Implement IPFS protocol as a URI scheme (not the entire multiaddr spec): `ipfs://<hash>`.
2. Implement the multiaddr protocol outside of the URI scheme (which would mean there wouldn't need to be URI scheme prefixes `i://`).

The first option is the simplest, but this would effectively mean that I'm not supporting the mutliaddr at all. Existing applications distributed on IPFS would contain links that are incompatible with URI, forcing these apps to be updated.

The second option would render links within existing IPFS applications compatible. Though, this would require a complete overhaul in how addresses and references are handled within a web application. This would include not only anchor elements, but every element and API containing a URI implementation. Not to mention that policies must be considered as well (same-origin policy). Further, all web applications depending on the URI scheme would be renders incompatible.

As I understand it, the intentions behind multiaddr (and IPFS) is to maintain a smooth transition from the existing models. How is mutliaddr intended to be implemented that is aligned with this intention?
